### PR TITLE
Deprecated `rust_toolchain.rustc_srcs`

### PR DESCRIPTION
**Pull request by UebelAndre**
Instead a [rust_analyzer_toolchain](https://github.com/bazelbuild/rules_rust/blob/0.12.0/rust/private/rust_analyzer.bzl#L214-L224) can be used.

### CODE REVIEW

File: rust/repositories.bzl [hunk 1]
```diff
@@ -586,31 +586,35 @@ def rust_repository_set(
...
         name (str): The name of the generated repository
         version (str): The version of the tool among "nightly", "beta', or an exact version.
         exec_triple (str): The Rust-style target that this compiler runs on
-        include_rustc_srcs (bool, optional): Whether to download rustc's src code. This is required in order to use rust-analyzer support. Defaults to False.
-        allocator_library (str, optional): Target that provides allocator functions when rust_library targets are embedded in a cc_binary.
+        include_rustc_srcs (bool, optional): **Deprecated** - instead see [rust_analyzer_toolchain_repository](#rust_analyzer_toolchain_repository).
+        allocator_library (str, optional): Target that provides allocator functions when rust_library targets are
+            embedded in a cc_binary.
         extra_target_triples (list, optional): Additional rust-style targets that this set of
-            toolchains should support. Defaults to [].
```
Comment (krasimirgg) [inline rust/repositories.bzl:599]: why drop these defaults?

Comment (UebelAndre) [reply rust/repositories.bzl:599]: The text here is redundant. Defaults are neatly rendered in our docs

Comment (UebelAndre) [reply rust/repositories.bzl:599]: https://bazelbuild.github.io/rules_rust/flatten.html#rust_repository_set

Comment (krasimirgg) [reply rust/repositories.bzl:599]: Thanks!

**Review by krasimirgg — APPROVED** [review]
Looks good.

**Pull request merged**